### DISCUSSION TITLE
2D game development with Pixi.js (or something else)

### DISCUSSION THREAD

**Discussion by brianberns** [fable-compiler/Fable#3740] [Q&A]
Hi, I'm interested in developing a 2D game with Fable. Once upon a time, Pixi.js was apparently the preferred library, but I don't think the bindings have kept up to date. (I tried both [this](https://github.com/fable-compiler/fable-pixi) and [this](https://github.com/fable-compiler/fable-graphics/blob/master/bindings/fable-import-pixi/Fable.Import.Pixi.fs).)

Does anyone have any suggestions for the best approach to 2D game development with Fable 4? Thanks!

**Comment by MangelMaxime** [comment]
Hello @brianberns,

I don't think a lot of people create games in Fable which can explains the lack of binding.

The "best approach" to approach game development with Fable 4 is by choosing a JavaScript library that you like and create the bindings for it. 

That's kind of what anything needs to happens when anyone start using a new library / ecosystem with Fable. 

If easier at first you can give a try to using dynamic typing which will allows you to have bindings and just explore how F# can fit into building a game with it. https://fable.io/docs/javascript/features.html#dynamic-typing-proceed-with-caution

**Comment by MangelMaxime** [comment]
I didn't mention it, but to create binding with have https://fable.io/ts2fable/ which is like the legacy tool which helps do the jobs done.

And the new in town, [Glutinum](https://fable.io/blog/2024/2024-01-01-Glutinum_a_new_era.html) which is slowly starting to evolve and try to improve the bindings API (still early beta ATM).

If you have questions regarding how to write a binding you can find articles online about it and also https://fable.io/docs/javascript/features.html contains lot of information about what interop feature Fable offers.

- https://hashset.dev/article/18_let_s_write_fable_bindings_for_a_js_library
- https://medium.com/@zaid.naom/f-interop-with-javascript-in-fable-the-complete-guide-ccc5b896a59f

Some of the articles you will find are probably a bit old but the philosophy and most of API should have stay the same. Bindings generated for Fable 1 are still usable with Fable 4 today for examples.

**Reply by brianberns** [reply]
Thank you, this is helpful. I would suggest pulling down the [Fable graphics page](https://fable.io/fable-graphics/) if those bindings are no longer maintained.

**Reply by MangelMaxime** [reply]
I didn't even know of this page.

The repo is marked as archived but I also changed the setting for the pages. Hopefully, it will still have an effect.
